### PR TITLE
fix(Tooltip): adds esc listener

### PR DESCRIPTION
**Pull request by paschalidi**
Hello 👋 ! 

Closes https://github.com/IBM/carbon-components-react/issues/1764

Adds event listener so that Esc key closes `Tooltip` when state is `open`.

#### Changelog

**New**

Adds event listener so that Esc key closes `Tooltip` when state is `open`.

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -266,6 +266,8 @@ class Tooltip extends Component {
     requestAnimationFrame(() => {
       this.getTriggerPosition();
     });
+
+    document.addEventListener('keydown', this.handleEscKeyPress, false);
```
Comment (asudoh) [inline src/components/Tooltip/Tooltip.js]: Please clean-up this event listener on un-mount.

File: src/components/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -391,6 +393,13 @@ class Tooltip extends Component {
     }
   };
 
+  handleEscKeyPress = event => {
+    const { open } = this.state;
+    if (open && keyDownMatch(event, [keys.ESC, keyCodes.ESC, keyCodes.IEESC])) {
```
Comment (dakahn) [inline src/components/Tooltip/Tooltip.js]: Great job grabbing this util 👍

Comment (paschalidi) [reply src/components/Tooltip/Tooltip.js]: Thanks @dakahn, appreciate it!

File: src/components/Tooltip/Tooltip.js (hunk 1 above)
Comment (paschalidi) [reply src/components/Tooltip/Tooltip.js]: thanks for the quick review! Keep it up!

**Comment by paschalidi** [comment]
@asudoh is something holding us back here?

**Comment by asudoh** [comment]
@paschalidi Sorry for being late!

**Pull request merged**